code/modules/antagonists/heretic/structures/defiled_altar.dm

```diff
@@ -0,0 +1,151 @@
+		/obj/item/bodypart/leg/left,
+		/obj/item/bodypart/leg/right,
+		/obj/item/bodypart/head) // There is probably (read:definitely) a better way than this but i'm not sure how to do so.
+	///Check to see if it is currently being used.
+	var/in_use = FALSE
+	///How much favour does each item spawned uses.
+	var/favour_cost = 1
+
+/obj/structure/destructible/defiled_altar/attacked_by(obj/item/weapon, mob/living/user)
+	if(!iscarbon(user))
+		return ..()
+
+	if(!IS_HERETIC_OR_MONSTER(user))
+		uh_oh(user)
+		return TRUE
+
+	if(istype(weapon, /obj/item/codex_cicatrix) || istype(weapon, /obj/item/melee/touch_attack/mansus_fist))
+		playsound(src, 'sound/items/deconstruct.ogg', 30, TRUE, ignore_walls = FALSE)
+		set_anchored(!anchored)
+		balloon_alert(user, "[anchored ? "":"un"]anchored")
+		return TRUE
+
+	return ..()
+
+/obj/structure/destructible/defiled_altar/attack_hand(mob/user, list/modifiers)
+	. = ..()
+	if(!IS_HERETIC_OR_MONSTER(user))
+		uh_oh(user)
+		return TRUE
+
+	if(!IS_HERETIC(user))
+		balloon_alert(user, "your lowly form is rejected!")
+		return TRUE
+
+	var/datum/antagonist/heretic/heretic_datum = IS_HERETIC(user)
+	if(heretic_datum.favours < favour_cost)
+		balloon_alert(user, "no favours to call!")
+		return TRUE
+
+	INVOKE_ASYNC(src, PROC_REF(show_radial), user)
+	return TRUE
+
+/*
+ * Wrapper for show_radial() to ensure in_use is enabled and disabled correctly.
+ */
+/obj/structure/destructible/defiled_altar/proc/show_radial(mob/living/user)
+	in_use = TRUE
+	create_item(user)
+	in_use = FALSE
+
+/*
+ * Ask the user to choose between organs or bodyparts
+ * It then shows the user of radial of possible items based on the type they chose,
+ * user then can chose any items they would like to spawn.
+ */
+/obj/structure/destructible/defiled_altar/proc/create_item(mob/living/user)
+	// Assoc list of [name] to [image] for the radial
+	var/list/choices = list()
+	// Assoc list of [name] to [path] for after the radial, to spawn it
+	var/list/names_to_path = list()
+
+	// Assoc list of [name] to [image] for the radial, to choose between organs or bodyparts
+	var/list/organs_or_bodyparts = list()
+	organs_or_bodyparts["organs"] = icon('icons/obj/medical/organs/organs.dmi', "eyes")
+	organs_or_bodyparts["bodyparts"] = icon('icons/mob/human/bodyparts.dmi', "default_human_chest")
+
+	var/picked_type = show_radial_menu(
+		user,
+		src,
+		organs_or_bodyparts,
+		require_near = TRUE,
+		tooltips = TRUE,
+		)
+
+	switch(picked_type)
+		if("organs")
+			for(var/obj/item/organ/organs as anything in organs_list)
+				names_to_path[initial(organs.name)] = organs
+				choices[initial(organs.name)] = image(icon = initial(organs.icon), icon_state = initial(organs.icon_state))
+		if("bodyparts")
+			for(var/obj/item/bodypart/bodyparts as anything in bodyparts_list)
+				names_to_path[initial(bodyparts.name)] = bodyparts
+				choices[initial(bodyparts.name)] = image(icon = initial(bodyparts.icon), icon_state = initial(bodyparts.icon_state))
```

Comment:
Rather then writing code to loop over a list of type options twice, why not use this switch to decide which list to use, and then loop over that afterwards? cleaner